### DISCUSSION TITLE
Pybricks Web IDE: File Storage Location

### DISCUSSION THREAD

**Discussion by mshield5** [pybricks/support#1421] [Q&A] [answered]
Hello,
My team is transitioning from Spike to Pybricks.  I am trying to figure out a good way for our team to collaborate on code.  We plan to use GitHub for configuration control / collaboration.  I have not figured out how to point the Pybricks Web IDE to a clone for the source code.  Is this possible?

I have a workaround, but the import/export steps make it rather cumbersome.

- Grab the latest code from GitHub
    - git pull
- Sync Web IDE
    - zip pybricks program source code
    - import zip into Web IDE
- Do Development
- Commit Changes
    - export pybricks program source coe zip
    - extract into git clone
    - git commit -am "My amazing changes"
    - git push

Thanks!

**Comment by laurensvalk** [comment, marked as answer]
File synchronization or git integration isn't currently on our roadmap, but it does come up quite frequently, so maybe we should consider it.

Since you are planning to use Git, perhaps you could consider using [Pybricks with VS Code](https://pybricks.com/projects/tutorials/dev/tools/vscode/). Officially we support only the online editor, but advanced users may find the alternate approach useful too.

**Reply by mshield5** [reply]
Thank you for confirming.  We plan to move to vscode when possible, but right now my team is not ready for pure Python development, they will be using the block coding feature as they learn Python.

I really love the way you have implemented the block coding to show real-time updates to python in the side panel.  This is going to be a huge help as the team moves forward to pure python.  The design decision to save block coding files as functional python with the block model defined in a comment is brilliant.  This will allow us to reuse what was created using block coding when we transition to pure Python development.

Thank you!

**Reply by laurensvalk** [reply]
Thanks for your comment! I’m glad to see those design decisions shining through.

Another reason we did it that way, is so you can import functions between block projects, or even from other Python scripts. This could even be another fun way to help your team transition to Python, by doing it one function at a time.

**Reply by iamthefaker** [reply]
Just to clarify/confirm, are you saying that when you using the block coding in Pybricks, it gives you a view of the equivalent Python code along side every change you make in the block coding?

**Reply by dlech** [reply]
Yes.

**Reply by iamthefaker** [reply]
Are you then able to edit the python code and changes are reflected back onto the block code?

**Reply by dlech** [reply]
Nope, it is only one-way.

**Reply by iamthefaker** [reply]
Thanks for help. Sorry if this is the wrong place to ask, but if I want to be able to directly edit .py files used in pybricks, do I have use pybricksdev with VisualCode (or whatever) ... ie. I can't use Pybricks local editor, and have direct access to the .py files?

**Reply by dlech** [reply]
The Python code generated from blocks is read-only, but if you want to modify it, you should be able to copy and paste it to a new file. Python files not created from blocks can of course be edited.
